cpu/atmega2560/periph/gpio.c

```diff
@@ -102,6 +102,24 @@ int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
     else {
         _SFR_MEM8(_ddr_addr(pin)) &= ~(1 << _pin_num(pin));
         res = bit_is_clear(_SFR_MEM8(_ddr_addr(pin)), _pin_num(pin));
+
+        if (res == 0) {
```

Comment:
@danielamkaer Are you sure with this?

Macro bit_is_clear:

```Javascript
#define bit_is_clear(sfr,bit) \
	(!(_SFR_BYTE(sfr) & _BV(bit))) 
```

Test whether bit bit in IO register sfr is clear. This will return **non-zero** if the bit is **clear**, and a **0** if the bit is **set**. 

And we want to clear the bit, or am I wrong?